icu4c/source/tools/toolutil/writesrc.cpp

```diff
@@ -357,6 +358,69 @@ usrc_writeUCPTrie(FILE *f, const char *name, const UCPTrie *pTrie, UTargetSyntax
     usrc_writeUCPTrieStruct(f, line, pTrie, line2, line3, line4, syntax);
 }
 
+U_CAPI void U_EXPORT2
+usrc_writeUnicodeSet(
+        FILE *f,
+        const char *name,
+        const USet *pSet,
+        UTargetSyntax syntax) {
+    // ccode is not yet supported
+    U_ASSERT(syntax == UPRV_TARGET_SYNTAX_TOML);
+
+    fprintf(f, "name = \"%s\"\n", name);
```

Comment:
These two new functions should not take nor print the name strings. They should just print the set/map data.

Note that other such functions here are pretty narrow as well, and you already print another name *before* calling these functions. (And Elango suggested printing the short one first -- whether you agree with that or not, the caller should control that, and it should be visible in the call site.)

Reply:
d528e355493ed019e493c9c9858280cb71613a55